packages/nextra/src/locales.ts

```diff
@@ -0,0 +1,27 @@
+import { NextRequest, NextResponse } from 'next/server'
+
+const PUBLIC_FILE = /\.(.*)$/
+
+export function locales(request: NextRequest) {
+  const shouldHandleLocale =
+    !PUBLIC_FILE.test(request.nextUrl.pathname) &&
+    !request.nextUrl.pathname.includes('/api/') &&
+    !request.nextUrl.pathname.includes('/.nextra/')
+
+  if (shouldHandleLocale) {
+    let href = request.nextUrl.href || '/'
+    if (href.endsWith('/' + request.nextUrl.locale)) href += '/'
+    if (href.endsWith('/')) href += 'index'
+
+    const locale = '.' + request.nextUrl.locale
```

Comment:
When preflight request have a cookie `NEXT_LOCALE = en`,
`request.nextUrl.locale` is still same as the accept-language here.  ( nextjs bug ?)

```suggestion
const locale = '.' + (request.cookies['NEXT_LOCALE'] || request.nextUrl.locale)
```